### PR TITLE
qemu: implement bridge struct

### PR DESCRIPTION
**Pull request by devimc**

### CODE REVIEW

File: qemu/qmp.go [hunk 1]
```diff
@@ -629,8 +629,9 @@ func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string)
 // using the device_add command.  blockdevID should match the blockdevID passed
 // to a previous call to ExecuteBlockdevAdd.  devID is the id of the device to
 // add.  Both strings must be valid QMP identifiers.  driver is the name of the
-// driver,e.g., virtio-blk-pci, and bus is the name of the bus.  bus is optional.
-func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, bus string) error {
+// driver,e.g., virtio-blk-pci, bus is the name of the bus and addr is the address
+// where the device is plugged. bus and addr are optional.
+func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, bus string, addr string) error {
```
Comment (markdryan) [inline qemu/qmp.go]: You could just do

driver, bus, addr string

@mcastelino I wonder at what point we're going to need to provide a structure for the parameters to this function.  Are we likely to be adding more fields in the future?

Comment (devimc) [reply qemu/qmp.go]: bootindex is another field that can be specified, but for now we don't need it

File: qemu/qmp.go [hunk 2]
```diff
@@ -639,6 +640,9 @@ func (q *QMP) ExecuteDeviceAdd(ctx context.Context, blockdevID, devID, driver, b
 	if bus != "" {
 		args["bus"] = bus
 	}
+	if addr != "" {
```
Comment (markdryan) [inline qemu/qmp.go]: I'm not sure about this.  The name of the device address property is bus specific.  "addr" as used here is PCI specific.  As we can specify different buses it doesn't feel right to hardcode a PCI specific key name in here.  I guess the best thing to do would be to only set the address if the bus is specified and to use the bus type to figure out what key to use.  @mcastelino any thoughts on this?

Comment (mcastelino) [reply qemu/qmp.go]: @devimc can we generate the address based on PCI properties instead of using a string. That may help.

Comment (markdryan) [reply qemu/qmp.go]: Here's the document that concerned me

https://github.com/qemu/qemu/blob/master/docs/qdev-device-use.txt#L9

Comment (devimc) [reply qemu/qmp.go]: unfortunately in Q35 we have to specify the address (number between 0x0 - 0x1f) hence each device must have a unique address, probably the best way to address this conflict could be adding a new method to hot plug just PCI devices ```ExecutePCIDeviceAdd```

Comment (devimc) [reply qemu/qmp.go]: ```bus``` must be specified too, because we want to hot plug devices on pci(e) bridges

Comment (markdryan) [reply qemu/qmp.go]: Fair enough.

**Comment by devimc** [comment]
@mcastelino  @markdryan  changes applied

**Pull request merged**